### PR TITLE
Improve code by removing unnecessary curly braces

### PR DESCRIPTION
**Pull request by mage2pratik**
### Description (*)

Improve code by removing unnecessary curly braces.
### Fixed Issues (if relevant)

1. magento/magento2#18813: Add curly braces to indirect variables to make them clear to understand

### Manual testing scenarios (*)

N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @mage2pratik. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: dev/tests/functional/tests/app/Magento/Backend/Test/Block/Messages.php [hunk 1]
```diff
@@ -152,7 +152,7 @@ public function clickLinkInMessage($messageType, $linkText)
     public function isVisibleMessage($messageType)
     {
         return $this->_rootElement
-            ->find($this->{$messageType . 'Message'}, Locator::SELECTOR_CSS)
+            ->find($this->$messageType . 'Message', Locator::SELECTOR_CSS)
```
Comment (orlangur) [inline dev/tests/functional/tests/app/Magento/Backend/Test/Block/Messages.php]: Please fix these two occurrences (commit amend + force push).

File: dev/tests/functional/tests/app/Magento/Sales/Test/Block/Adminhtml/Order/Create/Items/ItemProduct.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public function getCheckoutData(array $fields, $currency = '$')
                 $item['selector'],
                 $item['strategy'],
                 $item['input']
-            )->{$this->actions[$key]}();
+            )->$this->actions[$key]();
```
Comment (orlangur) [inline dev/tests/functional/tests/app/Magento/Sales/Test/Block/Adminhtml/Order/Create/Items/ItemProduct.php]: Please introduce a variable (commit amend + force push)

File: lib/internal/Magento/Framework/ObjectManager/Code/Generator/Persistor.php [hunk 3]
```diff
@@ -410,7 +410,7 @@ protected function _getDoPersistEntityMethod()
      */
     protected function _getRegisterFromArrayMethod()
     {
-        $body = "\$entity = \$this->{$this->_getSourceFactoryPropertyName()}->create(['data' => \$data]);\n"
+        $body = "\$entity = \$this->\$this->_getSourceFactoryPropertyName()->create(['data' => \$data]);\n"
```
Comment (orlangur) [inline lib/internal/Magento/Framework/ObjectManager/Code/Generator/Persistor.php]: Here curly braces are essential, please just revert changes.

File: lib/internal/Magento/Framework/ObjectManager/Code/Generator/Repository.php [hunk 4]
```diff
@@ -500,7 +500,7 @@ protected function _getDeleteByIdMethod()
      */
     protected function _getRemoveMethod()
     {
-        $body = "\$this->{$this->_getSourcePersistorPropertyName()}->registerDeleted(\$entity);";
+        $body = "\$this->\$this->_getSourcePersistorPropertyName()->registerDeleted(\$entity);";
```
Comment (orlangur) [inline lib/internal/Magento/Framework/ObjectManager/Code/Generator/Repository.php]: Here (in this file) curly braces are essential, please just revert changes.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by orlangur** [comment]
@mage2pratik looks good, waiting for builds.

**Review by orlangur — CHANGES_REQUESTED** [review]
@mage2pratik there are some failing integration tests, so, code behavior was unintentionally changed - please fix it.

After all builds are green, please squash all changes into a single commit.

**Comment by sivaschenko** [comment]
Hi @mage2pratik , would you like to continue working on this PR's progress?

**Comment by orlangur** [comment]
@sivaschenko looks like no but we should wait one more day :)

**Comment by sidolov** [comment]
Hi @mage2pratik , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**